generator/src/lib.rs

```diff
@@ -2323,8 +2327,14 @@ pub fn generate_aliases_of_types(
                 return None;
             };
             let alias_ident = name_to_tokens(alias);
-            let tokens = quote! {
-                pub type #name_ident = #alias_ident;
+            let tokens = if has_lifetimes.contains(&alias_ident) {
+                quote! {
+                    pub type #name_ident<'a> = #alias_ident<'a>;
+                }
+            } else {
+                quote! {
+                    pub type #name_ident = #alias_ident;
+                }
```

Comment:
Nit, I prefer to use `quote!` with parentheses instead of curly brackets - and put them on a single line - if the lines are short.

```suggestion
                quote!(pub type #name_ident<'a> = #alias_ident<'a>;)
            } else {
                quote!(pub type #name_ident = #alias_ident;)
```

Might look weird with the semicolons though?